### ISSUE TITLE
Unintuitive f-join behavior

### ISSUE THREAD

**Issue by sindikat** [rejeep/f.el#21]
```
ELISP> (f-join "a" "/b")
"../../../../b"
ELISP> (cd "~")
"/home/sindikat/"
ELISP> (f-join "a" "/b")
"../../b"
```

`f-join` behaves strange when meets an argument starting with slash, depending on a current directory in Emacs. In Python, for example, `os.path.join` simply discards all previous arguments when meets an argument starting with slash. Do you think it is better to make `f-join` behave like Python's?

**Comment by Fuco1** [comment]
Came here to start this same issue, nice to see it is already discussed.

This behaviour is, well, total shit :)

    (f-join "foo/" "/bar/" "baz/") => "../../../../../bar/baz".

I would expect it to put the slashes where they are mising and make multiple slashes into one.  That is the behaviour one wants when getting random crap input from other functions (half the time they have slashes, half the time not etc...

If the final result contains some `..` sequences, it should only be made cannonical after everything is joined.

What concerns the relative or not, I would simply not handle it in this function.  Users can decide then if they want to get absolute or relative path with different functions. `f-join` should "smartly" join paths and that's it.

**Comment by rejeep** [comment]
@Fuco1 Yeah, it sucks! :)

It would be nice to include a fix for this before releasing [v0.18](https://github.com/rejeep/f.el/tree/v0.18-wip), which I would like to release as soon as possible.

If anyone have time to implement this that would be awesome. I can do it myself, but currently all my time goes on redoing my kitchen, so I hardly ever get any time over these days.

**Comment by sindikat** [comment]
FWIW, Haskell's path join behavior is identical to Python's. Code:

    import Control.Monad (forM_)
    import System.FilePath ((</>))
    printJoinBehavior :: IO ()
    printJoinBehavior = do
      forM_ xs (\x -> putStrLn $ show x ++ ": " ++ foldr1 (</>) x)
      where xs = [["path"],
                  ["/path"],
                  ["path", "to", "file"],
                  ["/path", "to", "file"],
                  ["/", "path", "to", "file"],
                  ["path", "/", "to", "/", "file"],
                  ["path", "/to", "file"],
                  ["path", "//to", "file"],
                  ["/path", "/to", "file"],
                  ["path", "to/", "file"],
                  ["path", "to//", "file"],
                  ["/path", "to/", "file"],
                  ["/"]]
    
Output:

    ["path"]: path
    ["/path"]: /path
    ["path","to","file"]: path/to/file
    ["/path","to","file"]: /path/to/file
    ["/","path","to","file"]: /path/to/file
    ["path","/","to","/","file"]: /file
    ["path","/to","file"]: /to/file
    ["path","//to","file"]: //to/file
    ["/path","/to","file"]: /to/file
    ["path","to/","file"]: path/to/file
    ["path","to//","file"]: path/to//file
    ["/path","to/","file"]: /path/to/file
    ["/"]: /
